app/services/proofing/base.rb

```diff
@@ -25,7 +25,7 @@ def optional_attributes(*optional_attributes)
       end
 
       def attributes
-        required_attributes.concat optional_attributes
+        [*required_attributes, *optional_attributes]
```

Comment:
`.concat` is a mutation (same as `+=`) and this was causing some test flakes because it was modifying the required attributes array, and the new specs that read these accessors were sometimes seeing extra attributes